streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessor.java

```diff
@@ -17,70 +17,143 @@
             if (shouldForward(value)) {
                 internalProcessorContext.forward(key, value);
             } // else skip
         } else {
-            throw new NotImplementedException();
+            buffer(key, value);
+            enforceConstraints(internalProcessorContext.streamTime());
         }
     }
 
-    private boolean shouldForward(final Change<V> value) {
-        return !(value.newValue == null && suppress.suppressTombstones());
+    @SuppressWarnings("unchecked")
+    private <T> Serde<T> castSerde(final Serde<?> untyped) {
+        return (Serde<T>) untyped;
+    }
+
+    private void buffer(final K key, final Change<V> value) {
+        final long time = definedRecordTime(key);
+        final ProcessorRecordContext recordContext = internalProcessorContext.recordContext();
+
+        final Bytes serializedKey = Bytes.wrap(keySerde.serializer().serialize(null, key));
+        final byte[] serializedValue = valueSerde.serializer().serialize(null, value);
+
+        buffer.put(time, serializedKey, new ContextualRecord<>(serializedValue, recordContext));
     }
 
     private long definedRecordTime(final K key) {
-        return suppress.getTimeDefinition().time(internalProcessorContext, key);
+        return timeDefinition.time(internalProcessorContext, key);
     }
 
-    @Override
-    public void close() {
+    private void enforceConstraints(final long streamTime) {
+        final long expiryTime = streamTime - suppressDurationMillis;
+        if (overCapacity() || buffer.minTimestamp() <= expiryTime) {
+
+            drainExpiredRecords(expiryTime);
+
+            if (overCapacity()) {
+                switch (bufferFullStrategy) {
+                    case EMIT:
+                        buffer.evictWhile(this::overCapacity, this::emit);
+                        return;
+                    case SHUT_DOWN:
+                        throw new StreamsException(String.format(
+                            "%s buffer exceeded its max capacity [%d/%d] records and [%d/%d] bytes",
+                            internalProcessorContext.currentNode().name(),
+                            buffer.numRecords(), maxRecords,
+                            buffer.bufferSize(), maxBytes
+                        ));
+                }
+            }
+        }
     }
 
-    @Override
-    public String toString() {
-        return "KTableSuppressProcessor{" +
-            "suppress=" + suppress +
-            ", keySerde=" + keySerde +
-            ", valueSerde=" + valueSerde +
-            '}';
+    private void drainExpiredRecords(final long expiryTime) {
+        buffer.evictWhile(() -> buffer.minTimestamp() <= expiryTime, this::emit);
     }
 
-    public static class NotImplementedException extends RuntimeException {
-        NotImplementedException() {
-            super();
+    private void emit(final KeyValue<Bytes, ContextualRecord<byte[]>> next) {
```

Comment:
nit: `next` -> `evictedRecord` or just `record` ?

Reply:
Good point. I called it `toEmit`.